src/i_video.c

```diff
@@ -1358,12 +1356,8 @@ static void SetVideoMode(void)
 
     if (argbbuffer == NULL)
     {
-        SDL_PixelFormatEnumToMasks(pixel_format, &bpp,
-                                   &rmask, &gmask, &bmask, &amask);
-        argbbuffer = SDL_CreateRGBSurface(0,
-                                          SCREENWIDTH, SCREENHEIGHT, bpp,
-                                          rmask, gmask, bmask, amask);
-        SDL_FillRect(argbbuffer, NULL, 0);
+        argbbuffer = SDL_CreateRGBSurfaceWithFormatFrom(
+                     NULL, w, h, 8, 0, SDL_PIXELFORMAT_ARGB8888);
```

Comment:
The `8` here from the original code was not really correct (my bad). It can be safely replaced with `0` (SDL ignores it)